spring-integration-core/src/main/java/org/springframework/integration/dsl/ConsumerEndpointSpec.java

```diff
@@ -170,15 +173,19 @@ public S requiresReply(boolean requiresReply) {
 	/**
 	 * @param sendTimeout the send timeout.
 	 * @return the endpoint spec.
-	 * @see AbstractReplyProducingMessageHandler#setSendTimeout(long)
+	 * @see AbstractMessageProducingHandler#setSendTimeout(long)
 	 */
 	public S sendTimeout(long sendTimeout) {
 		assertHandler();
-		if (this.handler instanceof AbstractReplyProducingMessageHandler) {
-			((AbstractReplyProducingMessageHandler) this.handler).setSendTimeout(sendTimeout);
+		if (this.handler instanceof AbstractMessageProducingHandler) {
+			((AbstractMessageProducingHandler) this.handler).setSendTimeout(sendTimeout);
+		}
+		else if (this.handler instanceof AbstractMessageRouter) {
+			// This should probably go on the RouterSpec, but we put it here for consistency
```

Comment:
Well, my point was about simple configurations based on Lambdas and expressions or method invocation in most cases. All those options (`sendTimeout`, `async`, `requestHandlerAdviceChain` etc.) are really optional and, on the other hand, they are the same for any `MessageHandler` impl. So, I decided to keep them close to the `endpoint` even if we allow to configure `MH` directly, like in your `route(new PayloadTypeRouter())` sample.

Although with that your sample we really come back to the duplicated and confused options issue. Since we can configure `sendTimeout()` on the `PayloadTypeRouter()` directly as well as on the `EndpointSpec`.

There is a solution like merge `RouterSpec` with the `ConsumerEndpointSpec`, like we did for the `BarrierSpec`, `CorrelationHandlerSpec`. That will eat a couple overloaded methods and should kill this our discussion.
I think the `ConsumerEndpointSpec` for the `AbstractMessageRouter` variant  should still be there any way.